Add initial documentation for using AWS Security Token Service (#290)
openshift/cloud-credential-operator: docs/sts.md

   ```
3. Extract the OpenShift install binary from the release image
   ```
   oc adm release extract --command=openshift-install $RELEASE_IMAGE
   ```
4. Create an install-config.yaml
   ```
   ./openshift-install create install-config
   ```
5. Make sure that we install the cluster in Manual mode
   ```
   echo "credentialsMode: Manual" >> install-config.yaml
   ``` 
6. Create install manifests
   ```
   ./openshift-install create manifests   
   ```
7. Create AWS resources using [sts-preflight](https://github.com/sjenning/sts-preflight) tool (you will need aws credentials with sufficient permissions). Below command will generate public/private ServiceAccount signing keys, create the S3 bucket, upload the OIDC config into the bucket, set up an IAM Identity Provider that trusts that bucket, an IAM Role to be used for installation (this role has admin rights), and create IAM Roles for each AWS CredentialsRequest. It will also dump the files needed by the installer in the `_output` directory
   ```
   ./sts-preflight create --infra-name <aws_infra_name> --region <aws_region> --credentials-requests-to-roles /path/to/credreqs.yaml/downloaded/in/step/2   
   ```

> 8. Currently, the [sts-preflight](https://github.com/sjenning/sts-preflight) tool does not put condition on the created IAM role to allow only specific namespace/ServiceAccountName from assuming the Role. To more properly restrict who can assume which Role we need to manually modify each role to achieve this. Please note that role name might be truncated if it is more than 64 characters

Reviewer: ```suggestion
8. Currently, the [sts-preflight](https://github.com/sjenning/sts-preflight) tool does not put a condition on the created IAM role to allow only specific namespace/ServiceAccountNames from assuming the Role. To more properly restrict which identities can assume which Role we need to manually modify each Role to achieve this. Please note that Role names might be truncated if they are more than 64 characters
```
